theano/tensor/tests/test_nlinalg.py

```diff
@@ -516,3 +518,29 @@ def test_numpy_compare(self):
             t_n = f(A[2][i])
             n_n = numpy.linalg.norm(A[2][i], A[3][i])
             assert _allclose(n_n, t_n)
+
+
+class TestLogAbsDet(unittest.TestCase):
+
+    def setUp(self):
+        utt.seed_rng()
+        self.op_class = LogAbsDet
+        self.op = logabsdet
+
+    def validate(self, input_mat):
+        x = theano.tensor.matrix()
+        f = theano.function([x], self.op(x))
+        out = f(input_mat)
+        svd_diag = numpy.linalg.svd(input_mat, compute_uv=False)
+        numpy_out = numpy.sum(numpy.log(numpy.abs(svd_diag)))
```

Comment:
Having 1e-8 difference isn't a problem. You can have that type of error just by changing the order of addition when you add a vector of floats.

If the utt.assert_allclose pass, the I would compare again numpy.slogdet. If not, we need to understand why as they are supported to be equivalent. Maybe the fix would be to raise the comparison tolerance.